### ISSUE TITLE
Improve Dynamic Token Rings by adding clipping methods to improve user experience

### ISSUE THREAD

**Issue by theripper93** [foundryvtt/foundryvtt#11501]
### User Experience

Currently, Dynamic Token Rings are close to unusable to the average user as they would need to make specifically clipped images for them, this suggestions proposes adding clipping methods to allow for arbitrary images to be seamlessly used with the Dynamic Token Ring feature.

# Practical example
I have played with the current shader a bit to create an example

### Default
![image](https://github.com/user-attachments/assets/afa6a5ac-0b2c-41ef-a0b1-3a77c2e5d5a1)

### Modified

![image](https://github.com/user-attachments/assets/2f45c544-7f63-4cf7-bf3f-bb1030ab38c6)

Hardcoded values have been used in the shader for this test, but obviously the appropriate variables should be used 

![image](https://github.com/user-attachments/assets/a70d6014-b358-409b-983e-f6fabf1af77e)

# Proposal

### None
The current default behaviour

### Half
Clips the radial area outside of the token ring for the bottom range of UVs in the token texture (as per example above). This would allow the use of any portrait style image with a transparent background with the Token Ring

### Full
Clips the whole texture outside of the token ring. This would allow using any portrait image with a background with the Token Ring

## Add UV Offset option

This option would allow to offset the token texture in the X\Y coordinates

**Comment by aaclayton** [comment]
Yeah, good suggestion. It's tricky because sometimes the part of the token that should "pop out" is towards the bottom of the texture, so even conservatively clipping a lower third might truncate some things that you would want visible. You ran into this with your example where truncating the lower half of the token would have chopped off part of its left arm.

**Comment by aaclayton** [comment]
@Feu-Secret for your consideration

**Comment by MrVauxs** [comment]
I will note that if it is possible for this to use image masks (for example, an additional "Masking Image" below the Subject Art configuration), this would make the introduction of non-circular tokens actually doable, which (IMO) the vast majority of popular tokens on the internet are.

One example is actually showcased in #11530, with the gears being inward instead of outward. Any tokens that are not specifically tailor-made for this token, will have their art either clip the gears, or visibly disappear below the gear-line. Below are also some more examples from Loot Tavern:

![Image](https://github.com/user-attachments/assets/690027fa-696f-4fbd-b391-97218a88a39f)

With the usage of B/W mask images, token vendors can simply direct their users to use the appropriate mask they want, or edit their own from the available roster (such as masked top, bottom, full circle, corners, etc.).
